### ISSUE TITLE
S3 Download / TransferManager->DownloadFile + WaitUntilFinished never returns

### ISSUE THREAD

**Issue by Markus87** [aws/aws-sdk-cpp#1231]
I observed the problem that processes hang during download and never timeout. (waited for days to let it complete)

The stacktrace of the process looks like this:
```
Thread A
	[External Code]
	[Inline Frame] std::_Cnd_waitX
	[Inline Frame] std::condition_variable::wait
	Aws::Transfer::TransferHandle::WaitUntilFinished
	...

Thread B
	[External Code]
	[Inline Frame] std::_Cnd_waitX
	[Inline Frame] std::condition_variable::wait
	[Inline Frame] std::condition_variable::wait
	[Inline Frame] Aws::Utils::ExclusiveOwnershipResourceManager<Aws::Utils::Array<unsigned char> *>::Acquire
	Aws::Transfer::TransferManager::DoDownload
	[External Code]

Thread C-L 
	[External Code]
	Aws::Http::WinHttpSyncHttpClient::DoSendRequest
	Aws::Http::WinSyncHttpClient::MakeRequestInternal
	Aws::Http::WinSyncHttpClient::MakeRequest
	Aws::Client::AWSClient::AttemptOneRequest
	Aws::Client::AWSClient::AttemptExhaustively
	Aws::Client::AWSClient::MakeRequestWithUnparsedResponse
	Aws::S3::S3Client::GetObject
	Aws::S3::S3Client::GetObjectAsyncHelper
	[External Code]
```

https://github.com/aws/aws-sdk-cpp/blob/b17788a3f18a2bd3e46ffe044d178ca46bddc075/aws-cpp-sdk-core/source/http/windows/WinHttpSyncHttpClient.cpp#L101
I did see in the source that the nSendTimeout is ignored(-1) and thought this could cause the problem.
Is there a reason why this timeout is not set?

**Comment by KaibaLopez** [comment]
Hi @Markus87,
I'm having trouble replicating this issue, could you activate the trace logs and provide a sanitized version? 
Also, we've been making a lot of changes and bug fixes for transfer manager, so there's a possibility that it was a bug that's been addressed in newer versions, could you tell me the version of the sdk you're using?

**Comment by Markus87** [comment]
Hello @KaibaLopez !
I forgot to mention it, sorry.

The version I use is v1.7.65 and the program runs on Windows. I had the problem before and updated to v1.7.65 in hope that this might fix the issue. After that the problem still happened and I gave up on finding a solution assuming that the problem is isolated to that specific system and environment. But it appears the issue is more common than I thought and that's why I opened this issue.

I was never able to reproduce this issue on my development machine.
I will get you the trace logs but it may take me some time.

About the **nSendTimeout** not being set, do you know why it is disabled?

**Comment by Markus87** [comment]
Thanks for explaining @KaibaLopez !
I got the log you were asking for. There was more before it, but this should be everything related to the download that never finishes. If you need more please let me know.

[download.log](https://github.com/aws/aws-sdk-cpp/files/3608742/download.log)

**Comment by KaibaLopez** [comment]
Hi @Markus87, I've tested the download on windows and still can't really replicate the issue... what you're saying about the nSendTimeout does make sense though, could you try changing that value  and see if that at least makes it so you get a time out?

**Comment by Markus87** [comment]
> 
> 
> Hi @Markus87, I've tested the download on windows and still can't really replicate the issue... what you're saying about the nSendTimeout does make sense though, could you try changing that value and see if that at least makes it so you get a time out?

Yes I can do that but this will take some time.

**Comment by Markus87** [comment]
Sorry for the long wait @KaibaLopez 

I tested it with the nSendTimeout set to 300000 but it seems it does not make any difference, the process is still stuck "downloading".
I do not know why this causes no timeout, but i am pretty sure it has something to do with how the proxy is set up.
For testing I configured Internet Explorer with the same proxy-settings and it shows a similar behaviour. The file for the download is found but on saving it the download never gets anywhere, yet Internet Explorer at least gives up after some time.
